### PR TITLE
Added AbstractMLP class

### PR DESCRIPTION
**Pull request by Kamulau**
Summary of changes below:
Added interface MLPInterface.
Added public abstract class AbstractMLP extends Ranker implements MLPInterface.
Added abstract class AbstractNode<E extends AbstractEdge>
Added abstract class AbstractEdge<N extends AbstractNode>
MLP now extends AbstractMLP.
ListNet now extends AbstractMLP.

As a result of these changes, much repetitive code is now gathered higher up.

Future goal: Have SortNet extend AbstractMLP.

### CODE REVIEW

File: src/main/java/org/ltr4l/nn/AbstractEdge.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * limitations under the License.
+ */
+
+package org.ltr4l.nn;
+
+/**
+ * Edge (link) in the network.
+ * Holds information about which nodes are connected, the weight between the nodes, and dw.
+ */
+abstract class AbstractEdge<N extends AbstractNode> { //Serializable?
```
Comment (kojisekig) [inline src/main/java/org/ltr4l/nn/AbstractEdge.java]: Let's declare this as public

Comment (Kamulau) [reply src/main/java/org/ltr4l/nn/AbstractEdge.java]: ok I will change it.

File: src/main/java/org/ltr4l/nn/AbstractNode.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Node in the network.
+ * Holds information regarding the edges (which nodes are connected), and based on that information
+ * the total input and output. Also contains information about Activation.
+ */
+public class AbstractNode<E extends AbstractEdge> {
```
Comment (kojisekig) [inline src/main/java/org/ltr4l/nn/AbstractNode.java]: Why isn't this abstract class? Though the name of the class has "abstract" but the class is not abstract seems strange to me.

Comment (Kamulau) [reply src/main/java/org/ltr4l/nn/AbstractNode.java]: Apologies, I meant for it to be abstract. I will change this too.

Comment (Kamulau) [reply src/main/java/org/ltr4l/nn/AbstractNode.java]: Will keep it a regular class and change the name.

**Pull request merged**